x-pack/plugins/security_solution/common/api/search_strategy/matrix_histogram/matrix_histogram.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
```

Comment:
more of a nit, but while not list these files as `search_strategy/[domain]/index.ts`?

Reply:
this was a suggestion from @marshallmain I belive, for readability maybe? - I would see these are indexes as well. @marshallmain wdyt?

Reply:
Recently I've started trying to use `index.ts` only for files that are re-exporting things because it's easy to end up with a ton of different open files in VS code all called `index.ts` and all with relevant implementations. Naming implementation files with descriptive names has made it easier (for me, anyway) to find and keep track of what I'm working on.